### PR TITLE
fix: Don't clear localstorage on auth issues

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

When we encounter an auth issue, we clear the storage. This means on a subsquent page load, posthog-js sees that there is no state for the toolbar and doesn't load it. It would be better if we only hide the toolbar when the user explicitly chooses this

## Changes

* Refactor some reducer state to be based on selectors
* Removed some very old legacy values
* Take control of the local storage item (its always in localstorage) and persist it when the token is cleared
* Only remove the localstorage item when we explicitly close the toolbar.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

* Mostly manually, forcing the various situations

### CODE REVIEW

File: frontend/__snapshots__/scenes-other-toolbar--heatmap--dark.png
Comment (pauldambra) [inline frontend/__snapshots__/scenes-other-toolbar--heatmap--dark.png:1]: ruh roh... no data showing

Comment (pauldambra) [reply frontend/__snapshots__/scenes-other-toolbar--heatmap--dark.png:1]:
```
{error: TypeError: Cannot read properties of undefined (reading 'endsWith')
    at toolbarConfigLogic.apiUR…, reducerKey: 'elementStats', actionKey: 'getElementStats'}
actionKey
: 
"getElementStats"
error
: 
TypeError: Cannot read properties of undefined (reading 'endsWith') at toolbarConfigLogic.apiURL (http://localhost:8000/static/toolbar.js?t=1711105200000:59045:38) at recomputationWrapper (http://localhost:8000/static/toolbar.js?t=1711105200000:24201:31) at memoized (http://localhost:8000/static/toolbar.js?t=1711105200000:24118:26) at dependenciesChecker (http://localhost:8000/static/toolbar.js?t=1711105200000:24209:46) at Object.memoized [as apiURL] (http://localhost:8000/static/toolbar.js?t=1711105200000:24118:26) at Object.apiURL (http://localhost:8000/static/toolbar.js?t=1711105200000:25960:42) at Object.get2 [as apiURL] (http://localhost:8000/static/toolbar.js?t=1711105200000:25819:47) at getElementStats (http://localhost:8000/static/toolbar.js?t=1711105200000:65567:38) at newListeners.<computed> (http://localhost:8000/static/toolbar.js?t=1711105200000:28322:36) at http://localhost:8000/static/toolbar.js?t=1711105200000:25516:30
reducerKey
: 
"elementStats"
[[Prototype]]
: 
Object
```


yep, heatmap broken in some way

**Review by pauldambra — APPROVED** [review]
👍 

heatmap logic broken in some way but other than fixing that 🚢

File: frontend/__snapshots__/scenes-other-toolbar--heatmap--dark.png
Comment (benjackwhite) [reply frontend/__snapshots__/scenes-other-toolbar--heatmap--dark.png:1]: Eugh hacky way we are using logics here. think I have a fix

Comment (pauldambra) [reply frontend/__snapshots__/scenes-other-toolbar--heatmap--dark.png:1]: > Eugh hacky way we are using logics here

every PR review where I change kea code 🤣

**Pull request merged**